### PR TITLE
frontend: manually initialize the router

### PR DESCRIPTION
**Pull request by asdine**
The frontend router is now initialized manually in the main function, making it easier to inject dependencies to the handlers.

### CODE REVIEW

File: cmd/frontend/internal/app/ui/router.go [hunk 1]
```diff
@@ -102,6 +102,14 @@ func Router() *mux.Router {
 	return uirouter.Router
 }
 
+// InitRouter create the router that serves pages for our web app
+// and assigns it to uirouter.Router.
```
Comment (ryanslade) [inline cmd/frontend/internal/app/ui/router.go:106]: Isn't this just moving the global state to uirouter.Router?

File: cmd/frontend/internal/app/ui/router.go [hunk 2]
```diff
@@ -183,21 +191,17 @@ func newRouter() *mux.Router {
 	return r
 }
 
-func init() {
```
Comment (ryanslade) [inline cmd/frontend/internal/app/ui/router.go:186]: 👍

**Review by ryanslade — APPROVED** [review]
LGTM, moving in the right direction 😄

File: cmd/frontend/internal/app/ui/router.go (hunk 1 above)
Comment (asdine) [reply cmd/frontend/internal/app/ui/router.go:106]: I initially tried this but It's not possible because of circular dependencies

**Pull request merged**